src/math/Legendre.cpp

```diff
@@ -7,14 +7,15 @@
 
 double NuTo::Math::Polynomial::Legendre(int n, double x, int k)
 {
-    auto Factorial = [=](int i) -> int { return std::tgamma(i + 1); };
+    auto Factorial = [](int i) -> int { return std::tgamma(i + 1); };
 
     std::vector<double> vals(n + 1, 0.);
     if (n >= k)
         vals[k] = Factorial(2 * k) / (std::pow(2, k) * Factorial(k));
     for (int i = k + 1; i <= n; i++)
     {
-        vals[i] = ((2. * i - 1.) * x * vals[i - 1] - (i - 1. + k) * vals[i - 2]) / (i - k);
+        double vals_i_minus_2 = i >= 2 ? vals[i - 2] : 0;
```

Comment:
Ok, I probably messed that up when changing the algorithm from recursive to this loop. Sad that this bug didn't  show up in the test.

Reply:
I thought about changing the access from `[i]` to `.at(i)` which does bounds checking. Then you would get an `std::out_of_bounds` or something. This is a _tiny_ bit slower, which should not matter at all.